### PR TITLE
Add function to find policies for a resource

### PR DESCRIPTION
**Pull request by johnwu96822**
* Add a function to find policies for a resource
* Switch to Go mod
* Fixed a couple of print format errors

- [x] @abdulchaudhrycoupa 
- [x] @supriya

### CODE REVIEW

File: manager/memory/manager_memory.go [hunk 1]
```diff
@@ -104,3 +104,18 @@ func (m *MemoryManager) FindRequestCandidates(r *Request) (Policies, error) {
...
+
+// FindPoliciesForResource returns candidates that could match the request object. It either returns
+// a set that exactly matches the request, or a superset of it. If an error occurs, it returns nil and
+// the error.
+func (m *MemoryManager) FindPoliciesForResource(r *Request) (Policies, error) {
+	m.RLock()
+	defer m.RUnlock()
+	ps := make(Policies, len(m.Policies))
+	var count int
+	for _, p := range m.Policies {
```
Comment (abdulchaudhrycoupa) [inline manager/memory/manager_memory.go:116]: @johnwu96822 doesn't the range include an index, do you need count?

File: manager/sql/databases.go [hunk 2]
```diff
@@ -148,6 +149,29 @@ var Migrations = map[string]Statements{
...
+			p.effect,
+			p.conditions,
+			p.description,
+			subject.template AS subject,
+			resource.template AS resource,
+			action.template AS action
+		FROM
+			ladon_policy AS p
+
+			INNER JOIN ladon_policy_subject_rel AS rs ON rs.policy = p.id
```
Comment (abdulchaudhrycoupa) [inline manager/sql/databases.go:164]: @johnwu96822 just curious why this is an inner join and the action is a left join

Comment (johnwu96822) [reply manager/sql/databases.go:164]: Good point, i will investigate.

File: manager/memory/manager_memory.go (hunk 1 above)
Comment (johnwu96822) [reply manager/memory/manager_memory.go:116]: Haha this is copied from their function. Didn't realize this. Thanks!

Comment (johnwu96822) [reply manager/memory/manager_memory.go:116]: Errr.... m.Policies is a map, not a slice. Hence it needed an index variable.

**Comment by johnwu96822** [comment]
@abdulchaudhrycoupa @supriya I've changed the find policies by subject/resource queries so that they return full policies. Previous implementation only returned the subject/resource queried.

**Comment by abdulchaudhrycoupa** [comment]
👍🏽

**Comment by supriya** [comment]
lgtm

**Comment by johnwu96822** [comment]
[lockit v4.9.3] merged via Lockit CLI on 2020-12-07 10:51:31.417306 -0800 PST m=+3.681488091

**Pull request merged**